### DISCUSSION TITLE
Order of x/y in `paddingTopLeft` and `paddingBottomRight`

### DISCUSSION THREAD

**Discussion by vincerubinetti** [Leaflet/Leaflet#9107] [Q&A] [answered]
I'm concerned that the order of x/y in the padding tuples is ambiguous. I was assuming it was `paddingTopLeft: [top, left]` due to its name, but in my application (see below), it doesn't seem to work quite as expected. Diving into the code, it seems like this tuple gets passed to a `toPoint` function to make it into a `Point`, whose constructor is `Point(x, y)`, suggesting that it's actually `paddingTopLeft: [left, top]`. So which is it?

---

```ts
  // default fit padding
  let padding = { top: 20, left: 20, bottom: 20, right: 20 };

  // make room for legends
  if (props.showLegends) {
    // increase padding based on corner legend panel dimensions
    const padCorner = (v: "top" | "bottom", h: "left" | "right") => {
      const { width, height } = getBbox(`.leaflet-${v}.leaflet-${h}`);
      if (height > width) padding[h] = Math.max(width, padding[h]);
      else padding[v] = Math.max(height, padding[v]);
    };
    padCorner("top", "right");
    padCorner("top", "left");
    padCorner("bottom", "left");
  }
  // i've tested that the resulting padding here works as expected
  console.log(padding);

  // this part doesn't fit quite right. and yes i'm using a tiny, fine-grain zoomSnap
  map?.fitBounds(bounds, {
    paddingTopLeft: [padding.top, padding.left],
    paddingBottomRight: [padding.bottom, padding.right],
  });
```

**Comment by IvanSanchez** [comment, marked as answer]
The code expects a `L.Point`. As you've seen, that's `[x,y]` (or `[left, top]`).

I think the naming confusion stems from the (old, old) way of remembering about the order of CSS rules (top-right-bottom-left). See e.g. https://stackoverflow.com/questions/356759/a-mnemonic-for-the-order-of-css-margin-and-padding-shorthand-properties . It sticks as much as lat-lng (see also https://macwright.com/lonlat/ ).

**Reply by vincerubinetti** [reply]
Thanks for the response and links, just wanted to confirm my findings. 

For maintainers: I really think it'd be worth it to explicitly say `[left, top]` right in the description for these padding properties to remove all doubt. I know it also says `Point` as the arg, but someone might not see it and it also requires going to another section of the docs. Even if they did see it,  the deliberate name of the padding prop could create doubt.

**Reply by IvanSanchez** [reply]
Feel free to send a PR with the suggested change.
